### ISSUE TITLE
Voice your opinion on VB support on MAUI (next evolution on Xamarin.Forms)

### ISSUE THREAD

**Issue by Happypig375** [dotnet/vblang#540]
https://github.com/dotnet/maui/issues/115#issuecomment-634447806
>vb.net is basically dead. I doubt it will even be supported in .net 7

Show them otherwise...

**Comment by Padanian** [comment]
No doubt vb needs to be here. It's the 5th or 6th most used dev lang in the world, for both front ends and back ends, for CLI and GUI.

**Comment by AdamSpeight2008** [comment]
I now think that Microsoft's implementation of Visual Basic on .Net (either framework or core) is dead.

**Comment by Nukepayload2** [comment]
@AdamSpeight2008 
I agree. Even WinUI 3 (a new Windows 10 desktop development workload) is dropping VB support.
https://github.com/microsoft/microsoft-ui-xaml/issues/2497#issuecomment-632368899

**Comment by vbcodec** [comment]
they don't do it, VB is legacy now

**Comment by Padanian** [comment]
I will then keep using VB.net on Winforms , maybe migrating to .net core 5. It's not for MS to push me  what tools and workloads.

**Comment by sharwell** [comment]
@Happypig375 I don't think one issue can capture the needs of both languages here. Can you file a separate issue related to VB support if that is important to you?

**Comment by Happypig375** [comment]
@sharwell Being mentioned in the _most liked issue_ is definitely better than opening an entirely new issue. Opening a new issue will result in significantly less visibility.

**Comment by MyBlueHorizon** [comment]
It cannot be denied that VB is a beginner-friendly language. The most important thing is that for some old VB GUI programs, this is an important support for cross-platform, and it can run cross-platform with few changes. And for some students, they may not need to write programs often afterwards. VB is a convenient language for them, and they still need these. This passage was translated into English by translation software, and there may be some errors.

**Comment by paul1956** [comment]
Is Maui the next version of Xamrin.Forms or WinForms or both?

**Comment by MyBlueHorizon** [comment]
> Is Maui the next version of Xamrin.Forms or WinForms or both or something else?

Xamarin.Forms
https://github.com/dotnet/maui

**Comment by cristianlt23** [comment]
What the Vb.net dialect lacks is a community that really loves it, saying that it likes Vb.net and developing in C # makes no sense.
I will continue to use VB.net until it is 60% of my code, as soon as another code is greater than 40% I change languages, but do you know how long this will take? For a long time, VB.net is stable, but I agree that you need an active community that will keep you evolved and ready for any project.

I would like to see VB.net be an IDE one day and change its name from VB.net to Eternal VB, it would be interesting to dispute the VB syntax style against the C syntax style and its variations (C ++, C #, Java ).


texto original em Português Brasil traduzido pelo google tradutor.

O que falta ao dialeto Vb.net é uma comunidade que realmente a ame, dizer que gosta do Vb.net e desenvolver em C# não faz sentido.
Eu vou continuar a utilizar o VB.net até que ele seja 60% do meu código, a partir do momento que outro código for maior que 40% eu mudo de idioma, mas sabe quanto tempo isso vai levar? muito tempo, o VB.net é estável, mas concordo que precise de uma comunidade ativa que o deixe sempre evoluído e pronto para qualquer projeto.

Eu gostaria de ver um dia o VB.net ser uma IDE e mudar de nome de VB.net para VB Eterno, seria interessante uma disputa entre o estilo VB sintaxe contra o estilo da sintaxe do C e suas variações (C++, C#, Java).

**Comment by cristianlt23** [comment]
No other technology that VB doesn't support like Blazor, Vazor or Xamarin will make such a difference, we can safely skip those tools.
If there is one tool that every developer, business, and government should strive to support in VB.NET, it is MAUI.

**Comment by DBenS** [comment]
VB is much more intuitive and faster to code creation than C# (and its terrible ";"). 

C# was a try of Microsoft to bring the people involved with "C" to its new language environment, with minor changes and traumas.

Who says VB "is for dummies",  does not know what code is running on the most advanced companies around the world - at least, for the time the language exists and the migration done from VB5 and VB6 to it. We must remember that C# was created in 2000 (and VB6 came in 1998). 

VB has been proving as reliable and fast as C# (since the ILASM is the same for both!). Microsoft simply discarding the VB language from any new environment is a big mistake.

**Comment by cristianlt23** [comment]
Honestly, I'm a little tired of Microsoft, they want to determine what to use by pushing C# on all developers, that's unfortunate!

I don't like C# and period.

Killing VB seems more like an attempt to put C# in a better position in the language rankings, but it won't beat JavaScript, Python and Java, 10 years from now it may not even be those languages ​​anymore.

I hope VB.NET can get out of Microsoft and find its own way. My fingers are crossed for that to happen!

meanwhile, what I can't do in VB I'm doing everything within the JavaScript ecosystem.

C# is not my choice.

**Comment by cristianlt23** [comment]
@DBenS 

That's cool, we're Brazilians hahaha.

as you said, I also hope that compilers and an IDE will be created for our VB mode.

**Comment by rrvenki** [comment]
https://download.cnet.com/Visual-Basic-VB-NET-Compiler/3000-20414_4-78387115.html Just saw this VB compiler for iOS. Looks like there are several other VB compilers for other platforms including multiplatform. Not all are free.

**Comment by lordporkbone** [comment]
At this point, if you want to keep using Visual Basic your best bet is [Mercury](https://www.remobjects.com/elements/mercury/).

**Comment by Echo-8-ERA** [comment]
> At this point, if you want to keep using Visual Basic your best bet is [Mercury](https://www.remobjects.com/elements/mercury/).

At USD$600/year?

**Comment by lordporkbone** [comment]
> > At this point, if you want to keep using Visual Basic your best bet is [Mercury](https://www.remobjects.com/elements/mercury/).
> 
> At USD$600/year?

Not ideal obviously, but if you’re making money from your work and you are adamant you’re not going to use any other language, it’s an option none-the-less.

**Comment by DBenS** [comment]
> > > At this point, if you want to keep using Visual Basic your best bet is [Mercury](https://www.remobjects.com/elements/mercury/).
> > 
> > 
> > At USD$600/year?
> 
> Not ideal obviously, but if you’re making money from your work and you are adamant you’re not going to use any other language, it’s an option none-the-less.

It's an option - thanks for sharing it.

But, as @cristianlt23 said, considering a complete MS free IDE (for "Community"), Mercury is not an option for standalone programmers.

And, since the MS compiler creates the same code and shares the same libraries for C# and VB.NET, and considering the base of programmers (and companies) developed for so many decades for VB, it would be a wrong option for Microsoft to obligate them to migrate.

This behavior will affect the confidence to Microsoft for long-term languages. Who could ensure that the same migartion wouldn't be performed with C# to any new language in few years ahead?

**Comment by chdcomputers** [comment]
I am coding on VB since 1989...

**Comment by lepete11** [comment]
There was a joke in the 90s when I was using FORTRAN77:  "50 years from now there will be a programming language, and it will be called FORTRAN."  Same thing for VB.NET, 50 years from now, VB.NET will be used.   I am sure the big banks (with a lot more money than MS) just love the idea of deprecating VB.NET.  

In the future, MS will be exculpating over their decision to deprecate VB.  In fact, I wouldn't want to be in the board room when a president says: "I want all the names of executive leadership who are responsible for killing VB."   

For those who commented on speed, it doesn't matter, since both C# and VB.NET converge to MSIL.

**Comment by cristianlt23** [comment]
I invite you to access:
1) [https://anthonydgreen.net/2022/12/07...ility/#respond](https://anthonydgreen.net/2022/12/07/path-to-sustainability/#respond)
2) https://gitter.im/VB-NET/community

and if you feel convinced, I invite you to visit

3) https://www.patreon.com/ThatVBGuy

Thanks for reading
